Lib/test/test_struct.py

```diff
@@ -652,6 +652,13 @@ def test_format_attr(self):
         s2 = struct.Struct(s.format.encode())
         self.assertEqual(s2.format, s.format)
 
+    def test_issue35714(self):
+        # Embedded null characters should not be allowed in format strings.
+        for s in '\0', '\144\u0064\000xf', 'd\0d', '>ih\0', '=Q\0\0':
```

Comment:
What is the difference between all these cases? Why they are needed?

Reply:
The embedded null is located at different positions of the string.

Reply:
And why it is handled differently depending on the position?

I would test with `'\0'`, `'2\0i'` and `b'\0'`.